.github/workflows/ci.yml

```
@@ -0,0 +1,113 @@
+        with:
+          dockerfile: Dockerfile
+
+  # Job 2: Static Analysis (Semgrep)
+  semgrep:
+    name: Semgrep Security Scan
+    runs-on: ubuntu-latest
+    container:
+      image: returntocorp/semgrep
+    if: (github.actor != 'dependabot[bot]')
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.10)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)